project/src/components/main-screen/main-screen.tsx

```diff
@@ -3,9 +3,26 @@ import Map from '../map/map';
 import CityList from '../citys/citys';
 import { useAppSelector } from '../../hooks';
 import EmptyListScreen from '../empty-list-screen/empty-list-screen';
+import SortForm from '../sortform/sort-form';
+import { useEffect, useState } from 'react';
+import { Offer } from '../../types/offers';
 
 function MainScreen(): JSX.Element {
   const { currentCity, offersInCity } = useAppSelector((state) => state);
+  const [activeCard, setActiveCard] = useState(-1);
+  const [selectedCard, setSelectedCard] = useState<Offer | undefined>(undefined);
+
+  const onHoverHandler = (locationId: number) => setActiveCard(locationId);
+
+  useEffect(() => {
+    offersInCity.find((offer) => {
+      if (offer.id === activeCard) {
+        setSelectedCard(offer);
+      } else if (activeCard === -1) {
+        setSelectedCard(undefined);
+      }
+    });
+  }, [activeCard, selectedCard]);
```

Comment:
find используется не по назначению и если не находим карточку лучше помещать null
```suggestion
  useEffect(() => {
    const foundCard = offersInCity.find((offer) => offer.id === activeCard)
    setSelectedCard(foundCard || null)
  }, [activeCard, selectedCard]);
```